src/__tests__/to-have-value.js

```diff
@@ -124,6 +124,48 @@ describe('.toHaveValue', () => {
     `)
   })
 
+  test('throws type information when the expected input (type=text) value has loose equality with received value', () => {
+    const {container} = render(`<input data-testid="one" value="8" />`)
+    const input = container.firstChild
+    let errorMessage
+    try {
+      expect(input).toHaveValue(8)
+    } catch (error) {
+      errorMessage = error.message
+    }
+
+    expect(errorMessage).toMatchInlineSnapshot(`
+      "<dim>expect(</><red>element</><dim>).toHaveValue(</><green>8</><dim>)</>
+
+      Expected the element to have value:
+      <green>  8 (number)</>
+      Received:
+      <red>  8 (string)</>"
+    `)
+  })
+
+  test('throws type information when the expected input (type=number) value has loose equality with received value', () => {
```

Comment:
I don't think it's currently considering the input type. Just that the expected and actual have loose equality.

Reply:
No it does not indeed. With `input (type=text)` and `input (type=number)` I simply wanted to differentiate the tests.
Maybe renaming it to `'throws with type information when the expected text input value has loose equality with received value'` and `... number input value...`, respectively would be a better option?

Reply:
Yep. Then we don't need separate ones for different input types.